tutorials/javaFxPart3.md

```diff
@@ -7,15 +7,20 @@
 
 Picking up from where we left off last tutorial, we have successfully achieved the desired layout. Now let’s make the application respond to user input.
 
-Rather than to do everything in one try, let’s iterate and build up towards our final goal.
+JavaFX has an _event-driven architecture style_. As such, we programmatically define _handler_ methods to execute as a response to certain _events_. "Events" refers to the [Event class](https://docs.oracle.com/javase/8/javafx/api/javafx/event/Event.html). Two of the most commonly used subclasses are [ActionEvent](https://docs.oracle.com/javase/8/javafx/api/javafx/event/Event.html) (such as a button fired or a key frame finished) and [InputEvent](https://docs.oracle.com/javase/8/javafx/api/javafx/event/Event.html) (such as a mouse click).
+All UI components have a set of methods to define handlers with respect to different events. Some of these methods are unique to the component, but most are inherited from the parent classes `Node`, `Control`, etc.
+For example, the [Button](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/Button.html) class inherits `setOnAction` from parent `ButtonBase` and `setOnMouseClicked` from `Node`. Either one fulfills the functionality of Duke.
```

Comment:
From above comment, this link may be replace with this too:

https://openjfx.io/javadoc/17/javafx.controls/javafx/scene/control/Button.html

Reply:
Thank you for your suggestion and for finding the correct links. I will replace them.